### ISSUE TITLE
`node/no-top-level-await` - rule is reported in in "private" packages even though the error message says it's only forbidden in published packages

### ISSUE THREAD

**Issue by DetachHead** [oxc-project/oxc#24795]
the error message for `no-top-level-await` says:

```
Top-level `await` is forbidden in published modules.
help: Move the `await` inside an `async` function, as ES modules with top-level `await` cannot be loaded with `require(esm)`
```

however my `package.json` contains this, meaning it's not published:

```json
{
    "private": true
}
```
see https://docs.npmjs.com/cli/v11/configuring-npm/package-json#private

i'm not sure whether the solution is for this rule to not be reported in private packages, or to just update the error message to remove the mention of published packages. but either way the current behavior is quite confusing
